src/service/Review.ts

```diff
@@ -0,0 +1,342 @@
+import {
+    CodeReview,
+    CodeReviewRelation,
+} from '@prisma/client';
+import {PrismaClient} from '@prisma/client';
+import {ReactionData} from '../bolt/types';
+import {getUserInfo} from '../bolt/utils';
+
+async function findOrCreateUser(slackUserId: string) {
+    const prisma = new PrismaClient();
+
+    let user = await prisma.user.findFirst({
+        where: {
+            slackUserId,
+        }
+    });
+
+    if (!user) {
+        const userInfo = await getUserInfo(slackUserId);
+
+        user = await prisma.user.create({
+            data: {
+                slackUserId,
+                displayName: userInfo.displayName,
+            }
+        });
+    }
+    return user;
+}
+
+async function setCodeReviewerStatus(codeReview: CodeReview & {reviewers?: CodeReviewRelation[]}, slackUserId: string, status: string) {
+    const user = await findOrCreateUser(slackUserId);
+    const prisma = new PrismaClient();
+
+    let relation = await prisma.codeReviewRelation.findFirst({
+        where: {
+            codeReviewId: codeReview.id,
+            userId: user.id,
+        }
+    })
+
+    if (!codeReview.reviewers) {
+        codeReview.reviewers = [];
+    }
+
+    if (!relation) {
+        relation = await prisma.codeReviewRelation.create({
+            data: {
+                status: status,
+                codeReviewId: codeReview.id,
+                userId: user.id,
+            }
+        })
+        codeReview.reviewers.push(relation);
+    } else {
+        await prisma.codeReviewRelation.update({
+            where: {
+                userId_codeReviewId: {
+                    userId: user.id,
+                    codeReviewId: codeReview.id,
+                }
+            },
+            data: {
+                status: status,
+            },
+        });
+        codeReview.reviewers.forEach((relation) => {
+            if (relation.userId === user.id) {
+                relation.status = status;
+            }
+        });
```

Comment:
This updates the codeReview object but it looks like no more is done with this object in this function. Is this in the right place?

Reply:
This was added to sync up the code with the db.   The update statement on line 69-79 is to update the that specific record.   Since we already have the records read in memory already, we can just search and update that to match without having to refresh it from the database.